### PR TITLE
Added linq task

### PR DESCRIPTION
**Pull request by AntonPlotnikau**
added ten samples for linq task (not with tests yet)

### CODE REVIEW

File: 4. LINQ/Task/Task/Task/LinqSamples.cs [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+				{
+					c.CustomerID,
+					c.CompanyName,
+					Turnover = c.Orders.Sum(o => o.Total),
+					FirstOrderDate = c.Orders.OrderBy(o => o.OrderDate).Select(o => o.OrderDate).FirstOrDefault()
+				}
+				)
+				.OrderBy(c => c.FirstOrderDate.Year)
+				.ThenBy(c => c.FirstOrderDate.Month)
+				.ThenByDescending(c => c.Turnover);
```
Comment (s-gusev) [inline 4. LINQ/Task/Task/Task/LinqSamples.cs]: sorting by customer name is missing

File: 4. LINQ/Task/Task/Task/LinqSamples.cs [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+		[Category("Task")]
+		[Title("Task 6")]
+		[Description("Specify all customers who have a non-digital postal code, or the region is not filled, or the operator code is not specified in the phone")]
+		public void Linq006()
+		{
+			var customers = dataSource.Customers
+				.Where(
+					c => (c.PostalCode != null && c.PostalCode.Any(p => p < '0' || p > '9'))
+					|| c.Region == null
```
Comment (s-gusev) [inline 4. LINQ/Task/Task/Task/LinqSamples.cs]: what if c.Region would be just empty?

File: 4. LINQ/Task/Task/Task/LinqSamples.cs [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+					|| c.Phone.First() != '('
```
Comment (s-gusev) [inline 4. LINQ/Task/Task/Task/LinqSamples.cs]: what if the phone will be null or empty?

File: 4. LINQ/Task/Task/Task/LinqSamples.cs [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+				ObjectDumper.Write($"{customer.CustomerID} {customer.CompanyName} {customer.FirstOrderDate}");
+			}
+		}
+
+		[Category("Task")]
+		[Title("Task 5")]
+		[Description("Do the previous task, but get the list sorted by year, month, customer turnover (from the maximum to the minimum), and the client's name")]
+		public void Linq005()
+		{
+			var customers = dataSource.Customers
```
Comment (s-gusev) [inline 4. LINQ/Task/Task/Task/LinqSamples.cs:143]: will this query work correctly for the customer without orders?

File: 4. LINQ/Task/Task/Task/LinqSamples.cs [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+				ObjectDumper.Write($"{customer.CustomerID} {customer.CompanyName} {customer.Orders.Max(o => o.Total)}");
+			}
+		}
+
+		[Category("Task")]
+		[Title("Task 4")]
+		[Description("Get a list of customers including the month of the year they became clients (consider client first order month as a required date)")]
+		public void Linq004()
+		{
+			var customers = dataSource.Customers
```
Comment (s-gusev) [inline 4. LINQ/Task/Task/Task/LinqSamples.cs:122]: will this query work correctly for the customer without orders?

File: 4. LINQ/Task/Task/Task/LinqSamples.cs [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+				ObjectDumper.Write($"Id: {customer.CustomerID} Company: {customer.CompanyName} Code: {customer.PostalCode} Region: {customer.Region} Phone: {customer.Phone}");
+			}
+		}
+
+		[Category("Task")]
+		[Title("Task 7")]
+		[Description("Group all products by category, inside - by stock, within the last group sort by cost")]
+		public void Linq007()
+		{
+			var products = dataSource.Products.OrderBy(p => p.Category).ThenBy(p => p.UnitsInStock).ThenBy(p => p.UnitPrice);
```
Comment (s-gusev) [inline 4. LINQ/Task/Task/Task/LinqSamples.cs]: you have to use Group here, not Order

File: 4. LINQ/Task/Task/Task/LinqSamples.cs [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+			foreach (var product in products)
+			{
+				ObjectDumper.Write($"Name: {product.ProductName} Category: {product.Category} UnitsInStock: {product.UnitsInStock} UnitPrice: {product.UnitPrice}");
+			}
+		}
+
+		[Category("Task")]
+		[Title("Task 8")]
+		[Description("Group the goods into groups cheap, average price, expensive")]
+		public void Linq008()
```
Comment (s-gusev) [inline 4. LINQ/Task/Task/Task/LinqSamples.cs:217]: don't see groups

File: 4. LINQ/Task/Task/Task/LinqSamples.cs [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+			foreach (var item in statistics)
+			{
+				ObjectDumper.Write($"City: {item.City} AverageProfitability: {item.AverageProfitability} AverageIntensity: {item.AverageIntensity}");
+			}
+		}
+
+		[Category("Task")]
+		[Title("Task 10")]
+		[Description("Make the average annual activity statistics of clients by month (excluding the year), statistics by year, by year and by month")]
+		public void Linq0010()
```
Comment (s-gusev) [inline 4. LINQ/Task/Task/Task/LinqSamples.cs:258]: the task was done incorrectly

**Review by s-gusev — CHANGES_REQUESTED** [review]

File: 4. LINQ/Task/Task/Task/LinqSamples.cs (hunk 2 above)
Comment (AntonPlotnikau) [reply 4. LINQ/Task/Task/Task/LinqSamples.cs]: Fixed

File: 4. LINQ/Task/Task/Task/LinqSamples.cs (hunk 5 above)
Comment (AntonPlotnikau) [reply 4. LINQ/Task/Task/Task/LinqSamples.cs:122]: Fixed

File: 4. LINQ/Task/Task/Task/LinqSamples.cs (hunk 4 above)
Comment (AntonPlotnikau) [reply 4. LINQ/Task/Task/Task/LinqSamples.cs:143]: Fixed

File: 4. LINQ/Task/Task/Task/LinqSamples.cs (hunk 3 above)
Comment (AntonPlotnikau) [reply 4. LINQ/Task/Task/Task/LinqSamples.cs]: Fixed

File: 4. LINQ/Task/Task/Task/LinqSamples.cs (hunk 6 above)
Comment (AntonPlotnikau) [reply 4. LINQ/Task/Task/Task/LinqSamples.cs]: Fixed

File: 4. LINQ/Task/Task/Task/LinqSamples.cs (hunk 7 above)
Comment (AntonPlotnikau) [reply 4. LINQ/Task/Task/Task/LinqSamples.cs:217]: Fixed

**Pull request merged**